test/unit/propMerger.test.js

```diff
@@ -0,0 +1,63 @@
+import { mergeProps } from '../../src/HOFs/propMerger'
+
+describe('mergeProps()', () => {
+  it('should merge the `src` object into the `destination` object', () => {
+    const src = { width: 100, height: 200, imgixParams: {ar: "1:2", dpr: 2} }
+    const destination = { }
+    const result = mergeProps(src, destination);
+
+    expect(result).toEqual(
+      { width: 100, height: 200, imgixParams: { ar: "1:2", dpr: 2 } }
+    )
+  })
+
+  it('should not overwrite destination values with source values', () => {
+    const src = { width: 100, height: 200, imgixParams: {ar: "1:2", dpr: 2} }
+    const destination = { width: 101 }
+    const result = mergeProps(src, destination);
+
+    expect(result).toEqual(
+      { width: 101, height: 200, imgixParams: { ar: "1:2", dpr: 2 } }
+    )
+  })
+
+  it('should overwrite destination that resolves to `null` if source exists',
+    () => {
+      const src = { width: 100, height: 200, imgixParams: { ar: "1:1" } }
+      const destination = { width: 101, height: null, imgixParams: { rot: 90 } }
+      const result = mergeProps(src, destination);
+
+      expect(result).toEqual(
+        { width: 101, height: 200, imgixParams: { ar: "1:1", rot: 90 } }
+      );
+    }
+  )
```

Comment:
Actually I'm not sure I think this is a good idea. I think it's pretty standard behaviour to have the destination overwrite the src if set to `undefined`. i.e this:

```js
expect(mergeProps({ height: 100 }, {})).toEqual({ height: 100 })

expect(mergeProps({ height: 100 }, { height: undefined })).toEqual({ height: undefined });
// or
expect(mergeProps({ height: 100 }, { height: undefined })).toEqual({});
```

Reply:
This ms 💯 , good callout.

Something like...

```js
  const newProps = { ...destination };
  const newPropKeys = Object.keys(newProps);

  for (const [k, v] of Object.entries(src)) {
    if (newPropKeys.indexOf(k) == -1 && v !== null) {
      newProps[k] = v;
    }
```

This will both ensure we add new k, v pairs to the `destination` and allow `destination` to overwrite `src` k, v pairs when set as null or undefined.